src/slave/containerizer/mesos/linux_launcher.cpp

```diff
@@ -53,17 +56,24 @@ namespace mesos {
 namespace internal {
 namespace slave {
 
+struct CgroupsInfo
+{
+  // Flag indicating whether cgroups v2 is used.
+  bool cgroupsV2;
+
+  Option<std::string> freezerHierarchy;
+
+  Option<std::string> systemdHierarchy;
```

Comment:
it would be good if we could explain when these are set, e.g. if cgroupsv1 then freezer is set, but when is systemd set and how should it be used?